stories/Hits.stories.js

```diff
@@ -12,33 +14,66 @@ stories
       <Hits />
     </WrapWithHits>
   ))
-  .add('with custom rendering', () => (
-    <WrapWithHits linkedStoryGroup="Hits">
-      <Hits hitComponent={Product} />
-    </WrapWithHits>
-  ))
+  .add('with custom rendering', () => {
+    function Product({ hit }) {
+      return (
+        <div>
+          <Highlight attribute="name" hit={hit} />
+          <p>
+            <Highlight attribute="type" hit={hit} />
+          </p>
+          <p>
+            <Snippet attribute="description" hit={hit} />
+          </p>
+        </div>
+      );
+    }
+    Product.propTypes = {
+      hit: PropTypes.object.isRequired,
+    };
+    return (
+      <WrapWithHits linkedStoryGroup="Hits">
+        <Hits hitComponent={Product} />
+      </WrapWithHits>
+    );
+  })
   .add('with Panel', () => (
     <WrapWithHits linkedStoryGroup="Hits">
       <Panel header="Hits" footer="Footer">
         <Hits />
       </Panel>
     </WrapWithHits>
-  ));
-
-function Product({ hit }) {
-  return (
-    <div>
-      <Highlight attribute="name" hit={hit} />
-      <p>
-        <Highlight attribute="type" hit={hit} />
-      </p>
-      <p>
-        <Snippet attribute="description" hit={hit} />
-      </p>
-    </div>
-  );
-}
-
-Product.propTypes = {
-  hit: PropTypes.object.isRequired,
-};
+  ))
+  .add('with Insights', () => {
+    const insightsClient = (method, payload) =>
+      action(`[InsightsClient] sent ${method} with payload`)(payload);
+    const ProductWithInsights = connectInsights(insightsClient)(Product);
+    function Product({ hit, insights }) {
+      return (
+        <div>
+          <Highlight attribute="name" hit={hit} />
+          <button
+            onClick={() =>
+              insights('clickedObjectIDsAfterSearch', {
+                eventName: 'Add to cart',
```

Comment:
What do you think about documenting event names with a convention like `ADD_TO_CART`? To me it's super weird to name them like actual sentences. It can get really confusing (special characters, language support, etc.).

Reply:
There are no special characters or language support specifics to worry about to my knowledge, they are simple strings that show as is, in the dashboard to the user.
They are also documented as full sentences with spaces in the REST API.
https://www.algolia.com/doc/rest-api/insights/

Reply:
I asked in the Insights channel and there are no conventions for now.

See: https://algolia.slack.com/archives/C0TCN3XU4/p1553510236249300.